Vic2ToHoI4/Source/OutHoi4/OutHoi4World.cpp

```diff
@@ -0,0 +1,549 @@
+	}
+
+	for (const auto& country: countries)
+	{
+		if (country.second->isGreatPower() && country.second->getCapitalState())
+		{
+			outputToCommonCountriesFile(allCountriesFile, *country.second);
+		}
+	}
+
+	for (const auto& country: countries)
+	{
+		if (!country.second->isGreatPower() && country.second->getCapitalState())
+		{
+			outputToCommonCountriesFile(allCountriesFile, *country.second);
+		}
+	}
+
+	allCountriesFile << "\n";
+	allCountriesFile.close();
+}
+
+
+void HoI4::outputColorsFile(const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting country colors";
+
+	if (!Utils::TryCreateFolder("output/" + outputName + "/common/countries"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/countries");
+	}
+
+	std::ofstream output("output/" + outputName + "/common/countries/colors.txt");
+	if (!output.is_open())
+	{
+		throw std::runtime_error("Could not open output/" + outputName + "/common/countries/colors.txt");
+	}
+
+	output << "#reload countrycolors\n";
+	for (const auto& country: countries)
+	{
+		if (country.second->getCapitalState())
+		{
+			outputColors(output, *country.second);
+		}
+	}
+
+	output.close();
+}
+
+
+void HoI4::outputNames(const namesMapper& theNames,
+	 const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting names";
+
+	std::ofstream namesFile("output/" + outputName + "/common/names/01_names.txt");
+	if (!namesFile.is_open())
+	{
+		throw std::runtime_error("Could not open output/" + outputName + "/common/names/01_names.txt");
+	}
+
+	namesFile << "\xEF\xBB\xBF"; // add the BOM to make HoI4 happy
+
+	for (const auto& country: countries)
+	{
+		if (country.second->getCapitalState())
+		{
+			outputToNamesFiles(namesFile, theNames, *country.second);
+		}
+	}
+}
+
+
+void HoI4::outputUnitNames(const std::map<std::string, std::shared_ptr<Country>>& countries)
+{
+	LOG(LogLevel::Info) << "\t\tWriting unit names";
+
+	for (const auto& country: countries)
+	{
+		if (country.second->getCapitalState())
+		{
+			outputToUnitNamesFiles(*country.second);
+		}
+	}
+}
+
+
+void HoI4::outputMap(const States& states,
+	 const std::map<int, StrategicRegion*>& strategicRegions,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting map info";
+
+	if (!Utils::TryCreateFolder("output/" + outputName + "/map"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/map");
+	}
+
+	std::ofstream rocketSitesFile("output/" + outputName + "/map/rocketsites.txt");
+	if (!rocketSitesFile.is_open())
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/map/rocketsites.txt");
+	}
+	for (const auto& state: states.getStates())
+	{
+		auto provinces = state.second.getProvinces();
+		rocketSitesFile << state.second.getID() << "={" << *provinces.begin() << " }\n";
```

Comment:
dumping the first province is fine for rockets? It's always there, right? Never an empty state by accident and then random memory in file? :)

Reply:
Corrected